src/components/Invoices/InvoiceDetail/InvoiceDetailHeaderMenu.tsx

```diff
@@ -0,0 +1,147 @@
+import { useCallback, useState } from 'react'
+import { Span } from '../../ui/Typography/Text'
+import { Button } from '../../ui/Button/Button'
+import { Menu as MenuIcon } from 'lucide-react'
+import { useInvoiceDetail, useInvoiceNavigation } from '../../../providers/InvoiceStore/InvoiceStoreProvider'
+import { DropdownMenu, MenuItem, MenuList } from '../../ui/DropdownMenu/DropdownMenu'
+import { UpsertInvoiceMode } from '../../../features/invoices/api/useUpsertInvoice'
+import { InvoiceStatus, type Invoice } from '../../../features/invoices/invoiceSchemas'
+import { InvoiceVoidModal } from '../Modal/InvoiceVoidModal'
+import { InvoiceWriteoffModal } from '../Modal/InvoiceWriteoffModal'
+import { InvoiceResetModal } from '../Modal/InvoiceResetModal'
+import { InvoiceRefundModal } from '../Modal/InvoiceRefundModal'
+
+enum InvoiceDetailHeaderMenuActions {
+  Edit = 'Edit',
+  Void = 'Void',
+  Refund = 'Refund',
+  Writeoff = 'Writeoff',
+  Reset = 'Reset',
+}
+
+type InvoiceActionModalType = Exclude<InvoiceDetailHeaderMenuActions, InvoiceDetailHeaderMenuActions.Edit>
+
+const availableActions: Record<InvoiceStatus, InvoiceDetailHeaderMenuActions[]> = {
+  [InvoiceStatus.Sent]: [
+    InvoiceDetailHeaderMenuActions.Edit,
+    InvoiceDetailHeaderMenuActions.Void,
+    InvoiceDetailHeaderMenuActions.Writeoff,
+  ],
+  [InvoiceStatus.PartiallyPaid]: [
+    InvoiceDetailHeaderMenuActions.Writeoff,
+    InvoiceDetailHeaderMenuActions.Reset,
+  ],
+  [InvoiceStatus.Paid]: [
+    InvoiceDetailHeaderMenuActions.Refund,
+    InvoiceDetailHeaderMenuActions.Reset,
+  ],
```

Comment:
Edge case: if you create an invoice but forget to set a line item or price so the value is 0, we no longer allow you to edit it, and resetting it does nothing! So now you're kinda stuck (and this was probably an accident). I have done this with our form accidentally where I hit save before getting line items right!

I think the right fix is probably a backend change to consider zero value invoices as sent instead of paid unless there's a payment. If it's really some special situation, you can always record a $0 payment to mark it as paid.